dsp/window/window_complex.go

```diff
@@ -79,8 +81,9 @@ func LanczosComplex(seq []complex128) []complex128 {
 // Spectral leakage parameters: ΔF_0 = 4, ΔF_0.5 = 1.33, K = 2, ɣ_max = -26.5, β = -6.
 func TriangularComplex(seq []complex128) []complex128 {
 	a := float64(len(seq)-1) / 2
-	for i := range seq {
-		seq[i] *= complex(1-math.Abs(float64(i)/a-1), 0)
+	for i, v := range seq {
+		w := 1 - math.Abs(float64(i)/a-1)
```

Comment:
The window is symmetric, isn't there some optimisation opportunity? we could avoid calling math.Abs.

Reply:
Yeah. They're all symmetric. Is it OK if we get to that later (we already have a perfect optimisation by using the `Values` type)?

Reply:
Sure.